### PR TITLE
Improve list files RESTful API providing more useful info

### PR DESCRIPTION
**Pull request by JiamingMai**
Improve list files RESTful API providing more useful info. This PR add the following info to the response JSON:
1. Formatted file size
2. Alluxio path
3. UFS path
4. Last modification timestamp

### CODE REVIEW

File: dora/core/server/worker/src/main/java/alluxio/worker/http/ResponseFileInfo.java [hunk 1]
```diff
@@ -20,18 +22,34 @@ public class ResponseFileInfo {
...
 
+  private final String mPath;
+
+  private final String mUfsPath;
+
+  private final long mLastModificationTimeMs;
+
   private final long mLength;
 
+  private final String mSize;
```
Comment (dbw9580) [inline dora/core/server/worker/src/main/java/alluxio/worker/http/ResponseFileInfo.java:33]:
```suggestion
  private final String mHumanReadableFileSize;
```

**Review by dbw9580 — APPROVED** [review]

File: dora/core/server/worker/src/main/java/alluxio/worker/http/ResponseFileInfo.java (hunk 1 above)
Comment (JiamingMai) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/ResponseFileInfo.java]: Updated.

File: dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java [hunk 2]
```diff
@@ -156,6 +156,7 @@ private HttpResponseContext doListFiles(HttpRequest httpRequest, HttpRequestUri
       for (URIStatus uriStatus : uriStatuses) {
         String type = uriStatus.isFolder() ? "directory" : "file";
         ResponseFileInfo responseFileInfo = new ResponseFileInfo(type, uriStatus.getName(),
+            uriStatus.getPath(), uriStatus.getUfsPath(), uriStatus.getLastModificationTimeMs(),
```
Comment (huanghua78) [inline dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java:159]: @JiamingMai @dbw9580 does this work when multi-mounts is enabled?

Comment (JiamingMai) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java:159]: > @JiamingMai @dbw9580 does this work when multi-mounts is enabled?

If multi-mounts is used, then user needs to specify the full path when requesting this RESTful API. For example, `curl -X GET http://localhost:28080/v1/files?path=s3a://test-bucket/`.

Comment (dbw9580) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java:159]: this is worker side code, all Alluxio-namespace URLs should already be resolved to UFS URLs, so it should also work with multi-mounts.

Comment (huanghua78) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java:159]: So, the result list of files will have different paths or UFS paths with that in FUSE.
Applications/Users have to manually do another translation from UFS path to the path in FUSE.
I am not sure if this is the expected behavior.

Comment (huanghua78) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java:159]: > this is worker side code, all Alluxio-namespace URLs should already be resolved to UFS URLs, so it should also work with multi-mounts.

Yes, I know that.

So, the user will have a tree of files on client side FUSE mount point.
But the user has to use a UFS full path to do such listing on this API. And the result {path, UFS path} does not automatically map to any file on client.
The purpose to do such a listing is to have a full list of some directory and then filter out some files to be used in next step.
If the result from this API can not be mapped directly to the client view of namespace, the user can not use such list to do further data processing.

Comment (dbw9580) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java:159]: then I think the user should access the restful api from a proxy server, which has the mount table and can convert UFS URLs back to Alluxio URLs.

**Comment by JiamingMai** [comment]
alluxio-bot, merge this please.

**Pull request merged**